rest-api/api/participant_counts_over_time_api.py

```diff
@@ -0,0 +1,102 @@
+
+DATE_FORMAT = '%Y-%m-%d'
+DAYS_LIMIT = 100  # provisional, per design doc
+
+
+class ParticipantCountsOverTimeApi(Resource):
+
+  @auth_required(HEALTHPRO)
+  def get(self):
+    self.hpo_dao = HPODao()
+
+    # TODO: After enrollment status is filterable,
+    # wire in 'organization', 'site', and 'enrollmentStatus'.
+    enrollment_status = request.args.get('enrollmentStatus')
+    awardee = request.args.get('awardee')
+    stratification = request.args.get('stratification')
+    start_date = request.args.get('startDate')
+    end_date = request.args.get('endDate')
+
+    params = {
+      'enrollment_statuses': enrollment_status,
+      'awardees': awardee,
+      'stratification': stratification,
+      'start_date': start_date,
+      'end_date': end_date
+    }
+
+    # Most parameters accepted by this API can have multiple, comma-delimited
+    # values.  Arrange them into lists.
+    for param in params:
+      value = params[param]
+      if param in ['start_date', 'end_date', 'stratification']:
+        params[param] = value.encode()
+        continue
+      if value is None:
+        params[param] = []
+      else:
+        params[param] = value.encode().split(',')
+
+    params = self.validate_params(params)
+
+
+  def validate_params(self, params):
+
+    start_date_str = params['start_date']
+    end_date_str = params['end_date']
+    enrollment_statuses = params['enrollment_statuses']
+    awardees = params['awardees']
+    stratification = params['stratification']
+
+    # Validate dates
+    if not start_date_str or not end_date_str:
+      raise BadRequest('Start date and end date should not be empty')
+    try:
+      start_date = datetime.datetime.strptime(start_date_str, DATE_FORMAT).date()
+    except ValueError:
+      raise BadRequest('Invalid start date: %s' % start_date_str)
+    try:
+      end_date = datetime.datetime.strptime(end_date_str, DATE_FORMAT).date()
+    except ValueError:
+      raise BadRequest('Invalid end date: %s' % end_date_str)
+    date_diff = abs((end_date - start_date).days)
+    if date_diff > DAYS_LIMIT:
+      raise BadRequest('Difference between start date and end date ' \
+                       'should not be greater than %s days' % DAYS_LIMIT)
+    params['start_date'] = start_date
+    params['end_date'] = end_date
+
+    # Validate awardees, get ID list
+    awardee_ids = []
+    for awardee in awardees:
+      awardee_id = get_awardee_id_from_name({'awardee': awardee}, self.hpo_dao)
+      if awardee_id == None:
+        raise BadRequest('Invalid awardee name: %s' % awardee)
+      awardee_ids.append(awardee_ids)
+    params['awardee_ids'] = awardee_ids
+
+    valid_enrollment_statuses = EnrollmentStatus.to_dict()
+    for enrollment_status in enrollment_statuses:
+      if enrollment_status not in valid_enrollment_statuses:
+        raise BadRequest('Invalid enrollment status: %s' % enrollment_status)
+
+    valid_stratifications = ['TOTAL', 'ENROLLMENT_STATUS', 'GENDER_IDENTITY',
```

Comment:
We probably should have an enum for this, too? See above for handling.

If not, definitely define constants for these strings somewhere.

Reply:
Done.